allennlp/nn/samplers/sampler.py

```diff
@@ -9,6 +9,10 @@ class Sampler(Registrable):
     """
 
     def __call__(
-        self, logits: torch.Tensor, num_samples: int = 1, with_replacement: bool = True
+        self,
+        log_probs: torch.Tensor,
+        perturbed_log_probs: torch.Tensor = None,
+        num_samples: int = 1,
+        with_replacement: bool = True,
```

Comment:
Would need to be changed for all the subclasses too.

Reply:
But sampling with replacement doesn't even make sense for Gumbel, right? So maybe we should just remove this argument entirely. It could be a parameter in the `__init__` method for samplers that it makes sense for.

Reply:
You can change the with/without replacement for the other samplers, but stochastic is always without replacement. I'll switch it to be in the Init method.